### ISSUE TITLE
Use git mv when moving files

### ISSUE THREAD

**Issue by lucasprag** [scrooloose/nerdtree#586]
Hi.

Is there a way to use git mv when moving files?

Thanks

**Comment by zhenjun115** [comment]
@Lucasosf I do not think so.There is no keymap for "git mv".

**Comment by Lucasosf** [comment]
But is it possible? Is it related to the nerdtree?

I have a problem losing history when moving files in the nerdtree. With the cursor under a file name, I press `m` and `m` to move the file.

I will like to use `git mv` instead of justing moving as always.

I would like to implement it, I think it's a plugin right?

**Comment by PhilRunninger** [comment]
Yes, it's possible, and a good use of the NERDTree plugin API. Look at my [nerdtree-bwipeout-plugin](https://github.com/PhilRunninger/nerdtree-bwipeout-plugin) to see how easy it is to do. I started to look at solving your question, but I can't continue on it right now. I tried to implement it by copying and modifying the `NERDTreeMoveNode` function in `nerdtree/nerdtree_plugin/fs_menu.vim`. The part that stumped me was formatting the `git mv` command out to the shell.
